instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteRefreshTest.kt

```diff
@@ -110,99 +102,69 @@ class RouteRefreshTest : BaseTest<EmptyTestActivity>(EmptyTestActivity::class.ja
     }
 
     @Test
-    fun expect_route_refresh_to_update_traffic_annotations_and_incidents() {
-        // Request a route.
-        val routes = requestDirectionsRouteSync(coordinates).reversed()
-
-        // Create an observer resource that captures the routes.
-        val initialRouteIdlingResource = RoutesObserverIdlingResource(mapboxNavigation)
-            .register()
-
-        // Set navigation with the route.
-        runOnMainSync {
-            mapboxNavigation.setRoutes(routes)
+    fun expect_route_refresh_to_update_traffic_annotations_and_incidents_for_all_routes() =
+        sdkTest {
+            val routeOptions = generateRouteOptions(coordinates)
+            val requestedRoutes = mapboxNavigation.requestRoutes(routeOptions)
+                .getSuccessfulResultOrThrowException()
+                .routes
+                .reversed()
+
+            mapboxNavigation.setNavigationRoutes(requestedRoutes)
             mapboxNavigation.startTripSession()
-            mockLocationReplayerRule.loopUpdate(
-                mockLocationUpdatesRule.generateLocationUpdate {
-                    latitude = coordinates[0].latitude()
-                    longitude = coordinates[0].longitude()
-                    bearing = 190f
-                },
-                times = 60
-            )
-            mapboxNavigation.registerRouteProgressObserver { routeProgress ->
-                logD(
-                    "progress state: ${routeProgress.currentState}",
-                    LOG_CATEGORY
-                )
-                logD(
-                    "progress duration remaining: ${routeProgress.durationRemaining}",
-                    LOG_CATEGORY
-                )
-            }
-        }
-
-        // Wait for the initial route.
-        val initialRoutes = initialRouteIdlingResource.next()
+            stayOnInitialPosition()
+            val routeUpdates = mapboxNavigation.routesUpdates()
+                .take(2)
+                .map { it.navigationRoutes }
+                .toList()
+            val initialRoutes = routeUpdates[0]
+            val refreshedRoutes = routeUpdates[1]
+
+            mapboxNavigation.routeProgressUpdates()
+                .filter { routeProgress -> isRefreshedRouteDistance(routeProgress) }
+                .first()
+            mapboxNavigation.roadObjectsOnRoute()
+                .filter { upcomingRoadObjects ->
+                    upcomingRoadObjects.size == 2 &&
+                        upcomingRoadObjects.map { it.roadObject.id }
+                            .containsAll(listOf("11589180127444257", "14158569638505033"))
+                }
+                .first()
 
-        // Wait for the route refresh.
-        val refreshedRoutes = initialRouteIdlingResource.next()
-        initialRouteIdlingResource.unregister()
+            assertEquals(
```

Comment:
```suggestion
            assertEquals(2, requestedRoutes.size)
            assertEquals(
```

Reply:
added, but at the beginning as it doesn't make sense to carry on testing if the we have unexpected data

Reply:
let's check everything at the end of tests method if it's possible.

Reply:
Why do you want to check it at all?

In my understanding it's just a sanity check to make sure that testing data from resources won't unexpectedly change. Given the goal it doesn't make sense to wait for the end of the test to fail because testing data isn't the one we expected.